module/behaviour/strategy/KeyboardWalk/src/KeyboardWalk.cpp

```diff
@@ -77,64 +120,166 @@ namespace behaviour {
                     case KEY_UP: lookUp(); break;
                     case KEY_DOWN: lookDown(); break;
                     case 'q': quit(); return;
-                    default: log("Unknown Command");
+                    default:
+                        log<NUClear::ERROR>("Unknown Command");
+                        printStatus();
+                        break;
+                }
+            });
+
+            on<Trigger<LogMessage>, Sync<KeyboardWalk>>().then([this](const LogMessage& packet) {
```

Comment:
The Sync<KeyboardWalk> is what is causing this segfault, you can't use a reschedule operation here.